File: IKeep.Lib/Models/Relations/ElementGenericChore.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace IKeep.Lib.Models
{
    public class ElementGenericChore : Entity
    {
        public Guid ElementId { get; set; }
        [JsonIgnore]
        public virtual Element Element { get; set; }

        public string ElementName
        {
            get
            {
                return Element == null ? "" : Element.Name;
            }
        }
        public Guid GenericChoreId { get; set; }
        [JsonIgnore]
        public virtual GenericChore GenericChore { get; set; }
        public string GenericChoreDescription
        {
            get
            {
                return GenericChore == null ? "" : GenericChore.Description;
            }
        }
        public Status Status { get; set; }
    }

    //public enum Status
    //{
    //    Inactive,
    //    Active
    //}
}

File: IKeep.Web/ClientApp/src/models/ElementGenericChore.js
import Entity from './Entity'

class ElementGenericChore extends Entity
{
    constructor(json)
    {
        super(json);
        if(json)
        {
            this.ElementId = json.elementId;
            this.GenericChoreId = json.genericChoreId;
            this.Status = json.status;
            this.ElementName = json.generiElementName;
            this.GenericChoreDescription = json.genericChoreDescription;
        }
        else
        {
            this.ElementId = "";
            this.GenericChoreId = "";
            this.Status = 1;
            this.ElementName = "";
            this.GenericChoreDescription = "";
        }
        
    }
}

export default ElementGenericChore;